client-course-schedulizer/public/dataInterfaces.tsx

```diff
@@ -0,0 +1,60 @@
+enum Semester {
+  Fall = "FA",
+  Interim = "IN", // TODO: Remove?
+  Spring = "SP",
+  Summer = "SU", // TODO: Is this a thing?
+}
+
+enum Half {
+  First = "first",
+  Full = "full",
+  Second = "second",
+}
+
+interface Location {
+  building: string;
+  roomCapacity: number;
+  roomNumber: string;
+}
+
+interface Meeting {
+  // Like "MWThF" (all days on which the given Meeting time and room is applicable)
+  days: string;
+  // In minutes (usually 50)
+  length: number;
+  location: Location;
+  // Like "8:00AM" or "12:30PM"
+  startTime: string;
+}
+
+interface Instructor {
+  firstName: string;
+  lastName: string;
+}
+
+interface Course {
+  facultyHours: number;
+  name: string;
+  number: string;
+  prefix: string[];
+  studentHours: number;
+}
+
+interface Section {
```

Comment:
A way to signify async classes. No meeting time?

Reply:
Put a comment in 038a906. I think we just want to have those courses have an empty array of Meeting objects, and then if the registrar needs it to be printed as a field in the spreadsheet we can just extrapolate based on courses which have no meeting times. (I wonder how we will visualize these though. They will show up in the faculty loads tab, but I don't think they would appear anywhere else, so it might be difficult for a user to go back and edit them/know that they exist. Maybe we need a place for them to be shown in the UI?)

Reply:
Yeah, good idea. Maybe they will show up at the top of the schedule with no time on the UI.